### ISSUE TITLE
Losing Conversation History upon Login

### ISSUE THREAD

**Issue by liocalset** [wireapp/wire-webapp#688]
Hello,

Frequently in the past, and now every time when logging into the wire web app, I get this notification message:

> It’s the first time you’re using Wire on this device.
> For privacy reasons, your conversation history will not appear here.

The only thing is that it's not the first time I've used wire on the device. I thought maybe this was happening only after clearing Firefox's cache, but today I confirmed that all it takes is logging out and back in of wire for the above notification to show up. Upon logging in/getting the aforementioned notification message, all conversation history gets wiped.

As a test I tried logging in and out numerous times in a relatively quick succession, and upon each login I received the above notification message. After cliking "OK", conversation history from all past sessions are trashed.

System Specs:

OpenSUSE Leap 42.2
FireFox 50.1.0
wire-webapp Version 2017-01-23

Thanks!

**Comment by someoneEsle** [comment]
See also #261

**Comment by liocalset** [comment]
So the user has to tick the _"Remember Me"_ option for the conversation history not to get deleted upon logging out. I this default could be improved, because:

1. The _"Remember Me"_ option is not self-explanatory in what it actually does; it's actually quite obscure. In other words, _"You'll lose your conversation history every time you sign out unless you tick this box when you sign in"_ is what _"Remember Me"_ means.

2. Normal everyday users by default expect their conversation history to be saved without any additional intervention on their behalf. A by default temporary session certainly is more security focused and I always welcome that, but those that are wanting the additional privacy protection of what the _"Remember Me"_ option _not_ being enabled offers should have no trouble doing what I think is a better default:

I think a better default would be to **always** save conversation history, and the user could tick a box to log in with a temporary session instead of the other way around. Also, having to tick the _"Remember Me"_ box upon every login is a bit tedious.

Thank you.

**Comment by max-baz** [comment]
@liocalset would it solve the problem if _"Remember me"_ checkbox was enabled by default?

**Comment by gregor** [comment]
We have an internal ticket tracking usability issues in this area.

Please take a step back and consider that a "normal everyday user" does not exist as such. Wire is used by many different kind of "normal everyday users". These have expectations that can vastly divert from yours. Wire is focused on security foremost. We do not see security and privacy as trade offs to usabilty that require lots of compromises though. When designing and maintaining a user experience with a broad variety of users it is never possible to make everyone happy.

We consider the webapp as such as a client that people are less likely to use on a constant basis. If you do like Wire and use it daily, it is a lot more convenient to install the desktop apps for Windows, macOS and Linux. We do know that this is not possible in many office environments with out admin access to the machine though.

The web version of Wire is mainly intended fo a quick use on the go, on vacation, whereever you are. Therefore the default setting for the "Remember me" option is currently set to disabled to ensure we remove as much user data from the machine one logout as possible. The part of our everyday user base that is more focused on privacy (which tends to be the European crowd compared the US one) *expects* this kind of behavior.

That said - we will take your feedback into consideration.

**Comment by max-baz** [comment]
Maybe a simple and yet efficient solution would be to rename the label "Remember me" (which nobody understands) to "Forget my history after log out"?

**Comment by liocalset** [comment]
"Forget my history after log out" is a bit too long of a string IMO, but something similar to it such as "Incognito mode" I think makes sense.

The web app has a huge untapped market, and I for one am using it a lot more than just for temporary means. I believe it's currently good enough, too, to be used as such. Web apps on the desktop are the future if you look at current trends. All people use on desktop and laptop PC's in many cases these days is a web browser. Take a look at Gmail, Hangouts, Google Docs and all of Google's other products that run in a web browser. Hundreds of millions of people are using them, the enterprise included. I'm not saying C or C++ apps won't have their place also, as both can coexist peacefully into the distant future.

Time will tell on this one, but I know if I asked around in my circle eveybody would think the current default is kind of finicky. I would go as far as calling it a UI/UX "don't"; every other messaging app I can think of saves your conversation history by default, the Skype and Hangouts web apps included.

So, by getting the defaults and application features correct today, more users will come tomorrow. I don't think preemptively cutting yourself off from a huge market because of a (most likely small) percentage of Europeans wanting an offbeat default -- when they can easily just tick a box -- is the best choice for long-term market share growth in your sector. Sure, it may appease users with a certain use case, but to me it doesn't feel like the right default looking into the future.

And by the way, most all people on vacation are going to have their smartphone with them at all times. :smile:

**Comment by gregor** [comment]
@liocalset Thanks for your insights.

This just makes me wondering on how you come up with the "(most likely small) percentage of Europeans" using Wire? We are a European based company. Our development center is in Berlin, Germany. We have people from more than 20 different countries working on shaping the future of Wire. We are a messenger that heavily emphasizes privacy and security as core values of the product. While we compete for users with Skype and Hangouts to cater to their communication needs, this is what sets us apart. It is one of the reasons users prefer Wire over the competition. The big players in the market for the most part do not care about privacy or only see it as a marketing feature half baked added on top of the experience and not a core value of the product that is taken into consideration every time a new feature is added.

It is not part of their DNA. For us it is. We want to protect your digital privacy as good as we can. For some reasons people tend to believe that rules concerning privacy which have served our society in the analog world for a very long time should not apply to the digital world. We beg to differ. Your digital privacy is just as important. Privacy matters. This is a realisation that people in Europe have come to more quickly but the latest political developments on the other side of the Atlantic Ocean has led an increasing awareness in the US as well. Take a look at an overview of how different messaging platforms deal with security and privacy in this nice independent overview at https://www.securemessagingapps.com. 

Again, when looking at what most all people do, please take a step back and consider that the world hosts a wide variety of people. Having your smartphone with you does not help in remote places of the world, if you do not have roaming enabled. Charges for that in Europe still tend to be outrageous for example.  If no fallback wifi is available or hotels charge insane access fees, you are stuck with cheap internet cafes. Life with a smartphone is easy in developed countries, where internet access is available 24/7 everywhere at reasonable rates. Not all places of this world have this privilege.

I totally agree that webapps are a lot more powerful then they used to be. For Wire they already nicely coexist with native apps. Your offerings on desktop for macOS, Windows and Linux are essentially webapps. They are based on [Electron](http://electron.atom.io/), which is essentially a Chrome browser in a box. You are running the webapp - but the gain of some native integrations that the web cannot do. There you do not have to click a box. It is an installed app and you will always have a permanent client.

**Comment by fortran77** [comment]
I've found the current checkboxes slightly confusing, but instead of simply requesting that they be differently pre-checked, let me ask a broader question.

Who should be using Wire and when?

Should everybody be using Wire almost all the time? That would be ideal. It would make mass surveillance much harder. But then conversations should be saved by default in the web app, otherwise it will be extremely inconvenient for users who don't check the Remember Me box.

Or should we be using Wire only when we think we're being targeted, or only when we otherwise really need the privacy? In that case conversations should not be saved by default, to protect us from our own carelessness.

There is no way of resolving the above question in a simple manner. So, how do we make Wire do the right thing?

Answer: Let the user specify the type of privacy he needs, and let Wire provide that. Somewhere in the settings, there will be an option to choose between Privacy and Convenience. 

This is an easy choice for most people. Wire should then do the right thing automatically, and the user will not need to worry about whether to check Remember Me or not, or whether to request that all data be deleted on logout, if he checked Remember Me at login time.

The help text will say: “In Convenience mode, all your conversations will be saved on your current device. Nobody can eavesdrop on you via the network, but if they hack into or steal your computer, they can get a copy of your conversations.” And: “In Privacy mode, your conversations will be deleted every time you log out. When you log in again, your device will appear to be a new device, and it will not show you any old conversations. This is the mode to use if you want to leave no data behind on your computer. Then anybody hacking into your computer or stealing your computer will not  get access to your old conversations.”

Most people will opt for Convenience, **which is OK.** In the Convenience mode they won't need to complain about losing their conversations. They're still protected from eavesdropping on the network — and that is the greater threat for most of us.

Some of the time, some of the people will need the Privacy mode. _These people know who they are,_ so we can trust them to choose the right option.

We'll still want to provide the checkboxes so a manual override may be done, but they should be pre-checked to provide the desired Privacy or Convenience.

**Comment by andredasilvapinto** [comment]
This is happening to me all the time now. I keep on losing the conversation history on the desktop app. The mobile app works fine.

**Comment by andredasilvapinto** [comment]
Any idea of what might be causing this? It's always the same device, but for some reason wire thinks it is not (it shows several entries on the managed devices with different IDs when it is actually the same device). I am using macOS 10.12.4, Wire 2.14.2743. This seriously affects my user experience.

What does Wire use for determining the device fingerprint?

**Comment by andredasilvapinto** [comment]
I created a new issue in the wire-desktop repository.

**Comment by ZeiP** [comment]
Just in case someone stumbles across this issue, the issue is wireapp/wire-desktop#685.
